app/js/enhancedDate.js

```diff
@@ -1,5 +1,104 @@
 /* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+ * Module 2
+ * Author: Samson Negassi
+ */
+
+var enhancedDate = (function() {
+
+    //private 
+    var d = {};
+    var days = [ 'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 
+                'Friday', 'Saturday'];
+    var months = ["January" , "February" , "March" , "April", "May",
+        "June", "July", "August", "September", "October",
+        "November", "December"];
+    
+    //checks if d is an instansof Date and if not then set a date when
+    //other methods are invoked before setDate() method sets the date.
+    var checkAndSetDate = function () {
+        if (!(d instanceof Date)) {
+            d = new Date();
+        }
+    };
+        
+            
+    //public
+    return {
+        
+        setDate: function(arg = new Date()) {
```

Comment:
Unfortunately you can't do this in JavaScript yet. With ES6 you'll be able to set default arguments in this fashion, but for now, you need to just pass in arg and assign it a value inside the `setDate: function (arg) {`